HopsanGenerator/src/generators/HopsanExeGenerator.cpp

```diff
@@ -0,0 +1,375 @@
+#endif
+
+using namespace hopsan;
+
+HopsanExeGenerator::HopsanExeGenerator(const QString &hopsanInstallPath, const QString &compilerPath, const QString &tempPath)
+    : HopsanGeneratorBase(hopsanInstallPath, compilerPath, tempPath)
+{
+}
+
+
+
+bool HopsanExeGenerator::generateToExe(QString savePath, ComponentSystem *pSystem, const QStringList &externalLibraries, bool x64)
+{
+#ifdef _WIN32
+    if(mCompilerSelection.path.isEmpty())
+    {
+        printErrorMessage("Compiler path not specified.");
+        return false;
+    }
+#endif
+    // Set build directory
+    const QString buildPath = QString("%1/exe-build").arg(savePath);
+    // Clear old build artefacts
+    removeDir(buildPath);
+
+    // Create required directories
+    QDir saveDir(savePath);
+    saveDir.mkpath(buildPath);
+
+
+    //------------------------------------------------------------------//
+    //Obtain model name and version string
+    //------------------------------------------------------------------//
+
+    QString modelName = pSystem->getName().c_str();
+
+    //------------------------------------------------------------------//
+    //Copy HopsanCore files to export directory
+    //------------------------------------------------------------------//
+
+    if(!copyHopsanCoreSourceFilesToDir(buildPath)) {
+        printErrorMessage("Failed to copy Hopsan source code files.");
+        return false;
+    }
+    if(!copyDefaultComponentCodeToDir(buildPath)) {
+        printErrorMessage("Failed to copy default component library files.");
+        return false;
+    }
+
+    if(!copyExternalComponentCodeToDir(buildPath, externalLibraries, &mExtraSourceFiles)) {
+        printErrorMessage("Failed to export required external component library files.");
+        return false;
+    }
+    printMessage("Extra source files: "+mExtraSourceFiles.join(","));
+
+    //------------------------------------------------------------------//
+    //Copy source files from templates
+    //------------------------------------------------------------------//
+
+    bool c1 = copyFile(":/templates/exe_main.cpp", buildPath+"/exe_main.cpp");
+    bool c2 = copyFile(":/templates/exe_utilities.cpp", buildPath+"/exe_utilities.cpp");
+    bool c3 = copyFile(":/templates/exe_utilities.h", buildPath+"/exe_utilities.h");
+    if (!(c1 && c2 && c3)) {
+        printErrorMessage("Failed to copy template file(s)");
+        return false;
+    }
+
+    //------------------------------------------------------------------//
+    // Generate model file and export assets, replacing asset paths in model
+    //------------------------------------------------------------------//
+
+    QDir resourceDir(buildPath+"/resources/");
+    std::list<hopsan::HString> assets = pSystem->getModelAssets();
+    QMap<QString, QString> assetsMap;
+    if (!assets.empty()) {
+        printMessage("Exporting model assets");
+    }
+    for (const auto& asset : assets) {
+        QFileInfo assetInfo(asset.c_str());
```

Comment:
Is this block of code a copy from the fmu export code or is it specific to the exe? If copy, then a common utility function should be made.

Reply:
Good point, it is copied so I will make a utility function.